test/e2e/osa_test.go

```diff
@@ -83,4 +84,43 @@ var _ = Describe("Openshift on Azure e2e tests", func() {
 		})
 		Expect(err).ToNot(HaveOccurred())
 	})
+
+	It("should run the correct image", func() {
+		// image format is either:
+		// docker.io/openshift || registry.access.redhat.com/openshift3
+		format := strings.Split(c.cs.Config.Images.Format, "/")
+		prefix := fmt.Sprintf("%v/%v", format[0], format[1])
+
+		// these pods should have the same prefix
+		podsToCheck := map[string]struct{}{
+			"oauth-proxy":                    {},
+			"origin-control-plane":           {},
+			"origin-docker-registry":         {},
+			"origin-haproxy-router":          {},
+			"origin-node":                    {},
+			"origin-service-catalog":         {},
+			"origin-template-service-broker": {},
+			"origin-web-console":             {},
+			"prometheus-alert-buffer":        {},
+			"prometheus-alertmanager":        {},
+			"prometheus-node-exporter":       {},
+			"prometheus":                     {},
+		}
+
+		pods, err := c.kc.CoreV1().Pods("").List(metav1.ListOptions{})
+		Expect(err).ToNot(HaveOccurred())
+		for _, pod := range pods.Items {
+			// e2e check should ensure that no reg-aws images are running on box
+			Expect(strings.Contains(pod.Status.ContainerStatuses[0].Image, "registry.reg-aws")).ToNot(BeTrue())
+			parts := strings.Split(pod.Status.ContainerStatuses[0].Image, "/")
+
+			if len(parts) < 3 {
```

Comment:
@mjudeikis, would it be better to loop through the pods in my `podsToCheck` map and ensure their registry is correct?

If by some reason their images don't align with this structure:
```
repo/namespace/container
```
then we skip the test as the image format is unknown.  

Let me experiment with this.